app/Http/Controllers/ClassroomController.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Http\Requests\ClassroomRequest;
+use App\Models\Classroom;
+use App\Models\Mentor;
+use App\Services\ClassroomService;
+use App\Services\MentorService;
+use Illuminate\Http\Request;
+
+class ClassroomController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public ClassroomService $classroomService;
+    public MentorService $mentorService;
+
+    public function __construct(ClassroomService $classroomService, MentorService $mentorService)
+    {
+        $this->classroomService = $classroomService;
+        $this->mentorService = $mentorService;
+    }
+
+    public function index()
+    {
+        $classrooms = $this->classroomService->getAllClassroom();
+        return view('classroom.index', compact('classrooms'));
+    }
+
+    /**
+     * Show the form for creating a new resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function create()
+    {
+        $mentors = $this->mentorService->getAllMentor();
+        return view('classroom.add', compact('mentors'));
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param \Illuminate\Http\Request $request
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function store(ClassroomRequest $request)
```

Comment:
store and update method use 2 different request validate. For example, the update method needs 'id' of the model but store method doesn't

Reply:
fixed